cmd/cspi-mgmt/app/handler.go

```diff
@@ -254,3 +269,23 @@ func (c *CStorPoolInstanceController) removeFinalizer(cspi *apis.CStorPoolInstan
 		string(cspi.GetUID()))
 	return nil
 }
+
+// addPoolProtectionFinalizer is to add PoolProtectionFinalizer finalizer of cstorpoolinstance resource.
+func (c *CStorPoolInstanceController) addPoolProtectionFinalizer(cspi *apis.CStorPoolInstance) error {
+	// is CSPI is deleted or the finalizer is already present skip this step
```

Comment:
We can improve comment for this function. 
```
// addPoolProtectionFinalizer adds protection finalizer if it is not present in the object
```
Let us not talk about delete thing here as no such check is present in the function.